src/unix/aix.c

```diff
@@ -214,6 +215,20 @@ void uv__io_poll(uv_loop_t* loop, int timeout) {
   base = loop->time;
   count = 48; /* Benchmarks suggest this gives the best throughput. */
 
+  if (loop->flags & UV_LOOP_IDLE_TIME) {
+    user_timeout = timeout;
+    timeout = 0;
```

Comment:
Agreed this code is confusing, but after many attempts also turned out to be the implementation that was the least invasive. I'll try to explain:

1 - Backup the user's timeout value into `user_timeout`.
2 - Set `timeout = 0`.
3 - Run `epoll_wait` with `timeout == 0`.
4 - When it returns, set `timeout = user_timeout` and set `user_timeout = -2` to indicate that the first pass has already run.
5 - if `nfds > 0` then:
    5.1 - If an event is processed (meaning `w->cb()` is called) then end accumulating idle time
    5.2 - If a signal is received, and the user has a signal watcher set, then end accumulating idle time
6 - If after processing all `fd` and `nevents == 0` and the original `timeout == -1 || timeout > 0` then reenter `epoll_wait`
7 - If after processing all `fd` and `nevents > 0` then return immediately

Everything after step 4 happens anyway. So it just forces the first call to poll to return immediately to see if any events were waiting when poll was called (since there's no other way to determine if there were).

Reply:
@indutny Is this explanation clear? Since it's duplicated in all poll implementations I didn't add many inline comments. If you think that's necessary then I'll be happy to.

Reply:
It is almost clear. Could you add the `uv__metrics_update_idle_time` and `uv__metrics_set_provider_entry_time` to the list of your items?

What I'm worried about is that `uv__metrics_set_provider_entry_time` is called before `3`, and `uv__metrics_update_idle_time` is called before `6`. So when we get into `7` and eventually back to `3` the `entry` is zero. Does it sound right?

Reply:
@indutny I've redone the logic here. Using `user_timeout == -2` to control the loop was a stupid way of preventing the need to use one additional variable. I've changed the implementation and  hopefully made it more apparent. PTAL.

Reply:
Resolving this since the logic has since changed.